flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSink.java

```diff
@@ -121,7 +121,7 @@ public RMQSink(RMQConnectionConfig rmqConnectionConfig, SerializationSchema<IN>
 	 */
 	protected void setupQueue() throws IOException {
 		if (queueName != null) {
-			channel.queueDeclare(queueName, false, false, false, null);
+			channel.queueDeclare(queueName, true, false, false, null);
```

Comment:
This would just be icing on the cake but you could factor out these defaults into a package-private Util, with a method, something like `declareQueueDefaults(Channel channel, String queueName)`. Totally up to you, though. This is already an improvement.

If you do it, just do the refactor in a commit on top.

Reply:
I agree, thanks for the suggestion!